Slim/Plugin/Podcast/ProtocolHandler.pm

```diff
@@ -6,6 +6,57 @@ package Slim::Plugin::Podcast::ProtocolHandler;
 # version 2.
 
+=comment
+
+- how is image/icon/cover passed from the OPML item to the track 
+
+When a $url from an OPML item is added to current playlist, a $track object is created 
+and its image is obtained by calling handlerForUrl($url)->getMetadataFor($url). 
+If this protocol handler is HTTP, it needs to find the image just by using this $url, 
+there is no other information. XMLBrowser(s), when adding the url to the playlist, 
+call setRemoteMetadata which caches the OPML items's image link as "remote_image_$url". 
+Then the getMetadataFor uses this as one of the image sources. 
+Now, when playing the $track, a scanUrl is done and if there are redirections, the 
+$track->url is modified to $newurl which will then be used in further getMetadataFor calls.
+As this $newurl has no image's link entry in the cache, the "remote_entry_$url" is copied
+upon each redirection to "remote_image_newurl" during the scan. Ultimately, when the actual
+image is cached, getMetadataFor will get a local link to the image proxy
+
+- How scanUrl works
+
+For HTTP(S) protocol handler, the scanUrl calls Slim::Utils::Scanner::Remote::scanURL when
+a track starts to play to acquire all necessary informations. The scanUrl is called with 
+the $url to scan and a $song object in the $args. 
+That $song has a $track object that is created from the original $url set when OPML item 
+is added in the playlist. When scanning $url, the Slim::Utils::Scanner::Remote::scanURL
+creates a new $track object everytime the final URI returned by the GET is different from 
+the $url argument. This happens on HTTP redirection and/or if the $url argument differs 
+from $args->{'song'}->track->url. When it finally returns, scanUrl provides the final $track
+that is then replaced in the $song object and in the playlist
+The podcast plugin is using a "thin" type of protocol handler which simply encapsulate HTTPS(s).
+It is convenient to still capture some of the protocol handler's methods and quickly default
+to HTTP(S) built-in handling. Typically, the scanUrl de-encapsulate the $url into an HTTP(S)
+one (like myph://http://$url) and then relies on normal handling, but there are a few catches. 
+
+1- When a $track object has changed after scanUrl, the required protocol handler is re-evaluated
+and replaced by what matches the actual $track->url (there are many protocol handlers but that's
+too difficult to describe here). That means that the thin protocol handler would loose control
+if the $track->url has been reset to HTTP(S). To avoid that, an optional songHandler() method
+is called to see if the thin protocol handler still wants control.
+
+3- Now, not every thin protocol handler's methods will be called after this scan because some
+portion of LMS will always call handlerForUrl($track->url) and not use the protocol handler
+that is stored inside the $song object. This is a feature or a bug, your choice but for example, 

```

Comment:
Uh... oh... wow... `$song->currentTrackHandler()`. That indeed I didn't remember (even though I've modified some of it in the past). If there's inconsistency, then that's likely my fault. Looking at the uses of this I only see old commits by others. I'm pretty sure I always relied on `handlerForUrl()` instead...

Now the question is... should we fix the callers ignoring this? I guess you've got an impression of how much work that might be.

Reply:
hmmm .. there is this commit https://github.com/Logitech/slimserver/commit/1120bab6954664dad8bcb39a1c2bafec6ae36dac#diff-88d16f63d766ed2861633ed19e52ca3a2c244828808bd862673fd29f5f81ea5d done by a gangsta named @mherger :smile:

Joke aside, I'm terrified to try to "fix" these